src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1038,6 +1038,80 @@ let addOutputColumn (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColum
 
         loggingList
 
+let addBuildingBlock (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) (headerRange:Excel.Range) (selectedRange:Excel.Range) =
+
+    let rowCount = arcTable.RowCount + 1
+
+    let buildingBlockCells = Spreadsheet.CompositeColumn.toStringCellColumns newBB
+
+    let targetIndex =
+        let excelIndex = rebaseIndexToTable selectedRange headerRange
+
+        let headers = arcTable.ToExcelValues().[0] |> Array.ofSeq |> Array.map (fun header -> header.ToString())
+
+        if excelIndex < (float) (headers.Length - 1) then 
+
+            //We want to start looking after the chosen column in order to ignore the current column because it could be a main column
+            let headers = headers.[(int) excelIndex + 1..]
+
+            let mainColumNames =
+                // Get all cases of the union
+                Microsoft.FSharp.Reflection.FSharpType.GetUnionCases(typeof<CompositeHeader>)
```

Comment:
https://github.com/nfdi4plants/ARCtrl/blob/71a780f8df0290b0ede6d99858310521b439e2a2/src/Core/Table/CompositeHeader.fs#L159

Reply:
I like the idea!

Reply:
fixed